File: PYTHON/Pertemuan 01/JamTangan.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class JamTangan:
    def __init__(self, merek: str, model: str, water_resistant: bool):
        self.merek = merek
        self.model = model
        self.water_resistant = water_resistant
        self.jam = 7
        self.menit = 0

    def set_waktu(self, jam: int, menit: int):
        if 0 <= jam < 24 and 0 <= menit < 60:
            self.jam = jam
            self.menit = menit
            print(f"[Jam {self.merek}] Waktu diatur: {jam:02d}:{menit:02d}")

    def tampilkan_waktu(self):
        print(f"[Jam {self.merek}] Waktu sekarang: {self.jam:02d}:{self.menit:02d}")

    def aktifkan_alarm(self, jam_alarm: int, menit_alarm: int):
        print(f"[Jam {self.merek}] Alarm diset: {jam_alarm:02d}:{menit_alarm:02d} ⏰")

    def tampilkan_info(self):
        print(f"  Jam       : {self.merek} {self.model}")
        print(f"  Waktu     : {self.jam:02d}:{self.menit:02d}")
        print(f"  WR        : {'Ya' if self.water_resistant else 'Tidak'}")

File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/JamTangan.java
package id.ac.uph.pbo.pertemuan01;

public class JamTangan {
    private String merek;
    private String model;
    private int    jam;
    private int    menit;
    private boolean waterResistant;

    public JamTangan(String merek, String model, boolean waterResistant) {
        this.merek          = merek;
        this.model          = model;
        this.waterResistant = waterResistant;
        this.jam            = 7;
        this.menit          = 0;
    }

    public void setWaktu(int jam, int menit) {
        if (jam >= 0 && jam < 24 && menit >= 0 && menit < 60) {
            this.jam   = jam;
            this.menit = menit;
            System.out.printf("[Jam %s] Waktu diatur: %02d:%02d%n", merek, jam, menit);
        }
    }

    public void tampilkanWaktu() {
        System.out.printf("[Jam %s] Waktu sekarang: %02d:%02d%n", merek, jam, menit);
    }

    public void aktifkanAlarm(int jamAlarm, int menitAlarm) {
        System.out.printf("[Jam %s] Alarm diset: %02d:%02d ⏰%n",
                merek, jamAlarm, menitAlarm);
    }

    public void tampilkanInfo() {
        System.out.println("  Jam       : " + merek + " " + model);
        System.out.printf ("  Waktu     : %02d:%02d%n", jam, menit);
        System.out.println("  WR        : " + (waterResistant ? "Ya" : "Tidak"));
    }
}
